aclk/aclk_query.c

```diff
@@ -638,6 +663,39 @@ void aclk_query_threads_start(struct aclk_query_threads *query_threads)
     }
 }
 
+/**
+ * Checks and updates popcorning state of rrdhost
+ * returns actual/updated popcorning state
+ */
+
+ACLK_POPCORNING_STATE aclk_host_popcorning(RRDHOST *host)
+{
+    rrdhost_aclk_state_lock(host);
+    ACLK_POPCORNING_STATE ret = host->aclk_state.state;
+    if (host->aclk_state.state != ACLK_HOST_INITIALIZING){
+        rrdhost_aclk_state_unlock(host);
+        return ret;
+    }
+
+    if (!host->aclk_state.t_last_popcorn_update){
+        rrdhost_aclk_state_unlock(host);
+        return ret;
+    }
+
+    time_t t_diff = now_realtime_sec() - host->aclk_state.t_last_popcorn_update;
```

Comment:
Shouldn't timeouts be monotonic or something? This means that a date change will make the diff wrong or even negative. I realize that this is a widespread problem in the agent though.

Reply:
yep will change